tribe-ext-external-links-in-new-tab.php

```diff
@@ -264,54 +264,82 @@ public function get_type_option( $key = '' ) {
 		/**
 		 * Adds target="_blank" and rel="noopener noreferrer" to links in the content.
 		 * Skips anchor links and ones that already have a target set.
-		 * 
+		 *
 		 * @since 1.0.0
 		 *
 		 * @param string $content the post content.
 		 * @return string the modified post content.
 		 */
 		function open_content_links_in_new_tab( $content ) {
 			$pattern = '/<a(.*?)?href=[\'"]?[\'"]?(.*?)?>/i';
-		
-			$content = preg_replace_callback( 
-				$pattern, 
+
+			$content = preg_replace_callback(
+				$pattern,
 				function( $matches ) {
 					$tpl = array_shift( $matches ) ;
 					$href = isset( $matches[1] ) ? $matches[1] : null;
-			
+
 					// Ignore anchor links.
 					if ( trim( $href ) && 0 === strpos( $href, '#' ) ) {
-						return $tpl; 
+						return $tpl;
+					}
+
+					// Ignore local links.
+					if ( $this->is_local_link( $href ) ) {
+						return $tpl;
 					}
-			
+
 					// Ignore links that already have a target set.
 					if ( preg_match( '/target=[\'"]?(.*?)[\'"]?/i', $tpl ) ) {
 						return $tpl;
 					}
-			
-					return preg_replace_callback( 
-						'/href=[\'"]+(.*?)[\'"]+/i', 
+
+					return preg_replace_callback(
+						'/href=[\'"]+(.*?)[\'"]+/i',
 						function( $matches2 ) {
 							return sprintf( '%s target="_blank" rel="noopener noreferrer"', array_shift( $matches2 ) );
-						}, 
-						$tpl 
+						},
+						$tpl
 					);
-				}, 
-				$content 
+				},
+				$content
 			);
-		
+
 			return $content;
 		}
 
+		public function is_local_link( $url ) {
+			$url_components   = parse_url($url);
+			$local_components = parse_url( home_url() );
```

Comment:
use `wp_parse_url` and cache the result (in memory) since this doesn't change from link to link.